code/DynamicCache.php

```diff
@@ -92,14 +92,33 @@ protected function enabled($url)
         $basicAuthConfig = Config::inst()->forClass('BasicAuth');
         if($basicAuthConfig->entire_site_protected) {
             $member = null;
+
+            // NOTE(Jake): Required so BasicAuth::requireLogin() doesn't early exit with a 'true' value
+            //             This will affect caching performance with BasicAuth turned on.
+            if(!Security::database_is_ready()) {
```

Comment:
If `DB::is_active() === true` AND `Security::database_is_ready() === false` you should also skip the basic auth check, since your member tables aren't ready. This could occur when deploying the site for the first time.